### ISSUE TITLE
MSVC + CUB 1.15.0 + Windows.h compiler issues

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2#761]
Under MSVC, including `jitify.hpp` prior to `cub.cuh` results in compiler errors as of Thrust/CUB 1.15.0. 

This is not an issue with CUB <= 1.14, nor with GCC, nor if `cub.cuh` is included first.

With the complexity of our include hierarchies, makeing sure cub is only ever included before jitify is not going to be overly easy. 

For now, I'll be blocking CUB 1.15 on MSVC via CMake (as part of #758), but longer term this needs to be promoted to a CUB and / or Jitify issue (Ideally when we know which project is the root cause of the problem).

I've created a MWE of this which uses CMake to fetch the two dependencies and demonstrate via CI the issue, which can be used as reference in the promoted bug reports

https://github.com/ptheywood/jitify-cub-1.15-MSVC-demo

My next planned step (for when in windows and I have time) is to try and narrow down which part of the jitify headers is leading to the compiler error in CUB. Otherwise we could just open issues in both Nvidia Repos.

**Comment by Robadob** [comment]
That compile error makes it look like Jitify is leaking a define, if that can be tied down can be fixed on the Jitify end.

**Comment by Robadob** [comment]
Though possible jitify is including a system header which leaks the define?

**Comment by ptheywood** [comment]
Narrowed this down further to the following snippet, it's an NVRTC issue not jitify (when a define is non-zero).

```cuda
// -----
// Minimal snipped from Jitify which triggers the error.
#define NVRTC_GET_TYPE_NAME 1
#include <nvrtc.h>
// -----
#include <cub/cub.cuh>
int main(int artc, char* argv[]) {
    return EXIT_SUCCESS;
}
```

Updated my standalone repo a little although it needs some polish / change of focus to NVRTC, and set a larger CI sweep going to try and narrow if it's limited to specific NVRTC versions.

**Comment by ptheywood** [comment]
Reported to NVIDIA/CUB as https://github.com/NVIDIA/cub/issues/422

**Comment by ptheywood** [comment]
CUB/Thrust have now merged in patches for this, for Thrust/CUB 1.16.0. 

#758 is now set up to just not look for Thrust/CUB if using CUDA 11.6, assuming that 11.7 will ship with CUB/Thrust >= 1.16. 

CMake logic for finding CUB/Thrust, and if 1.15 was found downloading a corrected version was proving tempremental, so I stopped following that approach.
